crates/core/src/geoarrow/json.rs

```diff
@@ -428,78 +428,90 @@ fn set_column_for_json_rows(
     Ok(())
 }
 
-/// Creates JSON values from a record batch reader.
+fn set_geometry_column_for_json_rows(
+    rows: &mut [Option<JsonMap<String, Value>>],
+    array: Arc<dyn GeoArrowArray>,
+    col_name: &str,
+) -> Result<(), Error> {
+    for (i, row) in rows
+        .iter_mut()
+        .enumerate()
+        .filter_map(|(i, maybe_row)| maybe_row.as_mut().map(|row| (i, row)))
```

Comment:
Naive q - is there a way to zip the maybe_row and array element so we don't need to do the index based lookup the array value for each row?

Reply:
_Maybe_? It might be more of an ergonomic rather than functional change ... for now, I'll keep as-is b/c I'm just ape-ing the code I copied from **arrow**.